test/unit/org/apache/cassandra/config/ParameterizedClassExample.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.config;
+
+import java.util.Map;
+
+import org.junit.Assert;
+
+public class ParameterizedClassExample
```

Comment:
@tiagomlalves very small issue, this class can be inner in the respective test class it is used. No worries, I will move it there on merging.

Reply:
@smiklosovic I've started with that approach but noticed that returned list of constructors is different from what I expected as the outer class is always added as first argument of the constructor breaking the logic. This will end up with an exception stating that no suitable constructor was found. This rules out the usage of inner classes for Auth*.